docker_daemon/check.py

```diff
@@ -637,14 +638,46 @@ def _report_cgroup_metrics(self, container, tags):
                         if value is not None:
                             metric_func(self, mname, value, tags=tags)
 
+    def _build_network_mapping(self, container):
+        """Matches /proc/$PID/net/route and docker inspect to map interface names to docker network name"""
+        try:
+            proc_net_route_file = os.path.join(container['_proc_root'], 'net/route')
+
+            docker_gateways = {}
+            for netname, netconf in container.get(u'NetworkSettings').get(u'Networks').iteritems():
+                docker_gateways[netname] = struct.unpack('<L', socket.inet_aton(netconf.get(u'Gateway')))[0]
+
+            mapping = {}
+            with open(proc_net_route_file, 'r') as fp:
+                lines = fp.readlines()
+                for l in lines[1:]:
+                    cols = l.split()
+                    if cols[1] == '00000000':
+                        continue
+                    destination = int(cols[1], 16)
+                    for net, gw in docker_gateways.iteritems():
+                        if gw & destination == destination:
+                            mapping[cols[0]] = net
```

Comment:
it's true on physical networks but on first look, docker doesn't seem to allow it:
```
# docker network connect network1 twonets
Error response from daemon: service endpoint with name twonets already exists
```
But I'll look into it